src/com/xilinx/rapidwright/edif/EDIFLibrary.java

```diff
@@ -247,7 +247,7 @@ private void visit(EDIFCell cell, List<EDIFCell> visited, Map<String,EDIFCell> y
 		yetToVisit.remove(cell.getLegalEDIFName());
 		for(EDIFCellInst i : cell.getCellInsts()){
 			EDIFCell childCell = i.getCellType();
-			if(containsCell(childCell) && yetToVisit.containsKey(childCell.getLegalEDIFName())){
+			if(childCell.getLibrary() == this && containsCell(childCell) && yetToVisit.containsKey(childCell.getLegalEDIFName())){
```

Comment:
Is the `containsCell()` condition still necessary?

`containsCell()` eventually ends up here:
https://github.com/Xilinx/RapidWright/blob/96a79eb27b8f98d07ca3d8aa7726f80ca9153aaf/src/com/xilinx/rapidwright/edif/EDIFLibrary.java#L148-L149

thus the new condition `childCell.getLibrary() == this` catches the situation where a cell is attached to a library, but does not belong in that library's `cells` map?

Reply:
(I'm probing to see if there's a good reason for that latter case to appear to understand if we're just papering over the symptom and not the root cause)

Reply:
I think you are right, I don't know that `containsCell()` is still necessary.  The issue is that it was only checking by name if the cell was in the library.  Identically named cells from different libraries were causing an issue.  I will remove the second clause of the `if` statement.

Reply:
Ah, got it! The problem isn't that a cell wasn't present in the library it says it belongs to, but with identically named cells in two different libraries. As you were.